packages/api-explorer/src/JsonForm/configureJsonEditor.js

```diff
@@ -34,14 +38,16 @@ function setDefaultCustomization (JSONEditor) {
         }
 
         if (this.container !== undefined && this.title !== undefined) {
-          const classContainer = (key === 'upload' || key === 'base64') ? 'mia-container-upload-file-wrapper' : 'mia-container-wrapper' 
+          const classContainer = (key === 'upload' || key === 'base64') ? 'mia-container-upload-file-wrapper' : 'mia-container-wrapper'
           this.container.classList.add(classContainer)
         }
       }
     }
   });
 }
 module.exports = function configureJSONEditor(JSONEditor, intl, setFormSubmissionListener) {
+  if (JSONEditor.hasBeenEdited) return
```

Comment:
A bug was found, where it was seen that the decorations made to place the "types" were repeated "N" times where "N" was the number of endpoints that were opened.

In this screenshot we have opened 7 endpoint accordion before this.
<img width="188" alt="Schermata 2020-06-24 alle 17 05 06" src="https://user-images.githubusercontent.com/22480239/85702902-243a8a80-b6df-11ea-805f-5ee0cdf4eed1.png">

This is a repeated piece of code 
https://github.com/mia-platform/api-explorer/pull/78/files?file-filters%5B%5D=.css&file-filters%5B%5D=.js&file-filters%5B%5D=.snap#diff-0e249f966fe4875597330c391f2825b0R16

Reply:
oof, can you link here further information about what you said:

> A flag has been introduced to prevent the code from being rerun more than once.

Is there a PR you made to jsoneditor for this or is it something that is in progress on their side?

Reply:
Mia-platform forked Json-editor lib. No one pr is open. 
The code is written really badly. This makes managing any features almost impossible

Reply:
I don't think that it is a bug of json-editor.
JsonEditor does not deal with `endpoints that are opened`. Probably this bug is connected to some bad thing that we make to the instance or global-variable of JsonEditor

Reply:
Did you find the source of the problem?

Reply:
It could be the request animation frame, but I'm not sure

Reply:
I found out the problem.
In `setDefaultCustomization` function we get all the editors from `JSONEditor.defaults.editors` and we assign each of them to a class which extends the `JSONEditor.defaults.editors[i]`, in order that we can override some method of the class.

In the method `postBuild` we make `this.label.textContent = '${this.label.textContent} (${this.schema.type})'` in order to add the type.  Before doing this, we call `super.postBuild()`.

This works fine if `setDefaultCustomization` is called once.

What happens if we call `setDefaultCustomization` twice?
We get all the editors (which now each of them is our custom editor), we create a class that extends *our* custom class.
We call the super method, that add the `(type)` and now we add again the `(type)`.

This is why we need to add that flag to be sure that we extends the JSONEditor.editors only once

Reply:
We can solve it in a better way, changing the `setDefaultCustomization` class or to improve the form lifecycle?

Reply:
This flag is more a `isConfigured`. I'm fine with avoiding subsequent configurations of the same editor as long as there's no scenario where the subsequant calls to `configureJSONEditor` are used to modify the editor at runtime; if this is something that can happen I agree that we should improve `setDefaultCustomization` instead.

A way to fix this in the `setDefaultCustomization` could be introducing a flag inside the Customization class that is responsbile to avoid subsequent customization applies, this way during postBuild you will call the `super` method and stop further customizing.

Reply:
I created a singleton and isolated this configuration into one file. What do you think about it?